crates/contracts/solidity/Solver.sol

```diff
@@ -13,7 +13,11 @@ import { Trader } from "./Trader.sol";
 /// that proxies to some actual code. This way no solver can offer `ETH` from
 /// their private liquidity for the solution which could interfere with gas
 /// estimation.
-contract Solver {
+/// Because this contract code gets put at the address of a solver account it uses
+/// a custom storage layout to avoid storage slot conflicts with solver accounts
+/// that are smart contracts using the default layout.
+/// layout at uint256(keccak256("cowprotocol/services solver impersonator")) - 1
+contract Solver layout at 0x14f5b2c185fc03c75c787d1f0e10ea137cc6d235a0047448eff18c9a173a722a {
```

Comment:
Very cleaver use of the new feature!

In case the `Solver` implementation actually matters for the settlement, you might want to add fallback handling, similar to what the `Trader` does.

```solidity
address constant private SOLVER_IMPL = address(0x10001);

fallback() external payable {
    _fallback();
}

receive() external payable {
    _fallback();
}

function _fallback() private {
    bytes memory rdata = SOLVER_IMPL.doDelegatecall(msg.data);
    assembly { return(add(rdata, 32), mload(rdata)) }
}
```